pkg/partitionprune/prune.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package partitionprune
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/batch"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/partitionservice"
+	"github.com/matrixorigin/matrixone/pkg/pb/partition"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec"
+	p "github.com/matrixorigin/matrixone/pkg/sql/plan"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func Prune(
```

Comment:
All these functions need a comment.   It is NOT obvious to me

Reply:
Prune splits the input batch into multiple batches based on partition expressions.
It is used when PrimaryKeysMayBeModified, during DELETE operations, or when inserting data.
For each partition defined in the metadata, it evaluates the partition expression
and creates a new batch containing only the rows that match that partition's criteria.